rts/motoko-rts/src/region.rs

```diff
@@ -0,0 +1,49 @@
+use crate::memory::{ic::NEXT_REGION_ID, Memory};
+use crate::types::{size_of, Region, Value, Words, TAG_REGION};
+use crate::rts_trap_with;
+
+use motoko_rts_macros::ic_mem_fn;
+
+#[ic_mem_fn]
+pub unsafe fn region_new<M: Memory>(mem: &mut M) -> Value {
+    let r_ptr = mem.alloc_words(size_of::<Region>() + Words(1));
+    // NB. cannot use as_region() here as we didn't write the header yet
+    let region = r_ptr.get_ptr() as *mut Region;
+    (*region).header.tag = TAG_REGION;
+    (*region).id = NEXT_REGION_ID;
+    NEXT_REGION_ID += 1;
+    (*region).page_count = 0;
+    Value::from_ptr(region as usize)
```

Comment:
You also probably want to allocate a blob with initial capacity and assign it to the Value payload. In any case, that field has to have some valid Value in it (not necessarily a blob) for the GC to not get confused.